pkg/rebuild/rebuild/rebuildremote.go

```diff
@@ -60,36 +60,67 @@ var rebuildContainerTpl = template.Must(
 		"join":   func(sep string, s []string) string { return strings.Join(s, sep) },
 	}).Parse(
 		// NOTE: For syntax docs, see https://docs.docker.com/build/dockerfile/release-notes/
-		`#syntax=docker/dockerfile:1.4
-{{- if .UseTimewarp}}
-FROM gcr.io/cloud-builders/gsutil AS timewarp_provider
-RUN gsutil cp -P gs://{{.UtilPrebuildBucket}}/timewarp .
-{{- end}}
-FROM alpine:3.19
-{{- if .UseTimewarp}}
-COPY --from=timewarp_provider ./timewarp .
-{{- end}}
-RUN <<'EOF'
- set -eux
-{{- if .UseTimewarp}}
- ./timewarp -port 8080 &
- while ! nc -z localhost 8080;do sleep 1;done
-{{- end}}
- apk add {{join " " .Instructions.SystemDeps}}
- mkdir /src && cd /src
- {{.Instructions.Source| indent}}
- {{.Instructions.Deps | indent}}
-EOF
-RUN cat <<'EOF' >build
- set -eux
- {{.Instructions.Build | indent}}
- mkdir /out && cp /src/{{.Instructions.OutputPath}} /out/
-EOF
-WORKDIR "/src"
-ENTRYPOINT ["/bin/sh","/build"]
-`))
+		textwrap.Dedent(`
+				#syntax=docker/dockerfile:1.4
+				{{- if .UseTimewarp}}
+				FROM gcr.io/cloud-builders/gsutil AS timewarp_provider
+				RUN gsutil cp -P gs://{{.UtilPrebuildBucket}}/timewarp .
+				{{- end}}
+				FROM alpine:3.19
+				{{- if .UseTimewarp}}
+				COPY --from=timewarp_provider ./timewarp .
+				{{- end}}
+				RUN <<'EOF'
+				 set -eux
+				{{- if .UseTimewarp}}
+				 ./timewarp -port 8080 &
+				 while ! nc -z localhost 8080;do sleep 1;done
+				{{- end}}
+				 apk add {{join " " .Instructions.SystemDeps}}
+				 mkdir /src && cd /src
+				 {{.Instructions.Source| indent}}
+				 {{.Instructions.Deps | indent}}
+				EOF
+				RUN cat <<'EOF' >build
+				 set -eux
+				 {{.Instructions.Build | indent}}
+				 mkdir /out && cp /src/{{.Instructions.OutputPath}} /out/
+				EOF
+				WORKDIR "/src"
+				ENTRYPOINT ["/bin/sh","/build"]
+				`)[1:], // remove leading newline
```

Comment:
Should textwrap.Dedent auto-handle the leading newline, like in python?

Reply:
so python doesn't but the idiom is to include a backslash just after the open quotes to remove it. since there's no equivalent in golang string literals, we're a bit stuck. i'd like to obviate this but i'm also not crazy about diverging from the python API. i don't care very much either way, though

Reply:
Ah right, I forgot about the backslash idiom.

Either way seems good to me.